tests/cbmc/proofs/s2n_dh_params_free/s2n_dh_params_free_harness.c

```diff
@@ -30,8 +30,17 @@ void s2n_dh_params_free_harness()
     nondet_s2n_mem_init();
 
     /* Operation under verification. */
-    if (s2n_dh_params_free(dh_params) == S2N_SUCCESS) {
-        /* Postconditions. */
+    int result = s2n_dh_params_free(dh_params);
+
+    /* Postconditions. */
+    if (result == S2N_SUCCESS) {
         assert(dh_params->dh == NULL);
+    } else {
+        assert(dh_params == NULL && s2n_errno == S2N_ERR_NULL);
     }
+    
+    /* Cleanup before memory leak check.
+       1. free our heap-allocated `dh_params`, since `s2n_dh_params_free` only `free`s the contents.
+    */
```

Comment:
Nit: Generally in S2N we go with this block comment style:
```suggestion
    /* Cleanup before memory leak check.
     * 1. free our heap-allocated `dh_params`, since `s2n_dh_params_free` only `free`s the contents.
     */
```